lib/saluki-components/src/transforms/aggregate/mod.rs

```diff
@@ -776,6 +776,17 @@ where
                 forwarder.push(Event::Metric(new_metric)).await?;
             }
 
+            if hist_config.copy_to_distribution() {
+                let new_context = context.with_name(format!(
+                    "{}{}",
```

Comment:
They just append the prefix and name in the agent 
https://github.com/DataDog/datadog-agent/blob/main/comp/dogstatsd/server/server.go#L681

Reply:
We shouldn't be creating a new context if we don't have to. That's effectively what we're doing here, even if the prefix is the same... because `Context::with_name` will blindly clone the context's inner data before altering the name. It doesn't check if the name is actually change it not.